### ISSUE TITLE
Syntax highlighting boke after using a component which has a child with NVIM.

### ISSUE THREAD

**Issue by vighd** [vrischmann/tree-sitter-templ#56]
Hi!

With the latest version of this project, my syntax highlighting is failing when I using a component with another one inside. I seen a similar closed issue which one is should be fix this, but that fix is not working for me.

For test environment, I used the recommended [minimal](https://templ.guide/commands-and-tools/ide-support) NVIM configuration from the templ guide.

The current version of this project highlight my code like this:

![20240511_09h53m54s_grim](https://github.com/vrischmann/tree-sitter-templ/assets/33658509/d2128831-93f6-4a27-b568-6b924ac24b4f)

If I remove the @ character, the highlighting works again like this:

![20240511_09h54m28s_grim](https://github.com/vrischmann/tree-sitter-templ/assets/33658509/8b45b98b-cd50-4f2d-837d-b6baee740be3)

**Comment by vrischmann** [comment]
Hi,

a quick test show we don't handle import expressions using the form
```
@Foo{

}
```
right the now parser thinks the `{` starts a new block and it breaks parsing.

Can you share the code of `components.FieldControl` and ` components.Select` to make it easier for me to debug ?

**Comment by vighd** [comment]
Of course, here you go:

```
package components

type Select struct {
    Values      []SelectValues
    Class       string
    Selected    string
    Placeholder string
    Attrs       templ.Attributes
}

type SelectValues struct {
    Value string
    Data  string
}

templ (i Select) Select() {
    <div class="select is-fullwidth">
        <select { i.Attrs... }>
            for k, v := range i.Values {
                if k == 0 && i.Placeholder != "" {
                    <option selected?={ i.Selected=="" } disabled value="">{ i.Placeholder }</option>
                }
                <option selected?={ v.Value==i.Selected } value={ v.Value }>{ v.Data }</option>
            }
        </select>
    </div>
}
```

```
package components

templ FieldControl(label string) {
    <div class="field">
        <label class="label">{ label }</label>
        <div class="control">
            { children... }
        </div>
    </div>
}
```

**Comment by Odas0R** [comment]
> Hi,
> 
> a quick test show we don't handle import expressions using the form
> 
> ```
> @Foo{
> 
> }
> ```
> 
> right the now parser thinks the `{` starts a new block and it breaks parsing.
> 
> Can you share the code of `components.FieldControl` and ` components.Select` to make it easier for me to debug ?

Stumbled on this:

> right the now parser thinks the `{` starts a new block and it breaks parsing.

Is there any news?

**Comment by vrischmann** [comment]
Hi, I started working on this a while ago in this branch: https://github.com/vrischmann/tree-sitter-templ/tree/fix-import-expression-brace but IIRC it doesn't work and I didn't touch it since then.

I don't have time right now to work on this, if you want to work on this feel free to base your work on my branch.

**Comment by Odas0R** [comment]
Okay, will look into it when I have the chance :)

Thank you

**Comment by vrischmann** [comment]
Fixed with #82
